src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj

```
@@ -78,4 +78,23 @@
     </ProjectReference>
   </ItemGroup>
 
+  <Target Name="GenerateBuildMetadata" BeforeTargets="BeforeBuild">
+    <WriteLinesToFile
+      File="$(IntermediateOutputPath)Generated_BuildMetadata.cs"
+      Lines="// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```

```suggestion
      Lines="// <autogenerated />
```
We do not need license header for auto-generated intermediate files. For example, the intermediate files generated by source-generators do not have license header either. On the other hand, it should not hurt to tag it with `<autogenerated />`.